pkg/querier/querier.go

```diff
@@ -287,7 +291,7 @@ func (q querier) Select(_ bool, sp *storage.SelectHints, matchers ...*labels.Mat
 	// querying the long-term storage.
 	// Also, in the recent versions of Prometheus, we pass in the hint but with Func set to "series".
 	// See: https://github.com/prometheus/prometheus/pull/8050
-	if sp == nil || sp.Func == "series" {
+	if sp == nil || sp.Func == "series" && !q.queryStoreForLabels {
```

Comment:
I would make it more clear to read:
```suggestion
	if (sp == nil || sp.Func == "series") && !q.queryStoreForLabels {
```